libraries/src/Extension/ExtensionManagerTrait.php

```diff
@@ -88,7 +88,7 @@ public function bootModule($module, $applicationName): ModuleInterface
     public function bootPlugin($plugin, $type): PluginInterface
     {
         // Normalize the plugin name
-        $plugin = strtolower(str_replace('plg_', '', $plugin));
+        $plugin = strtolower(preg_replace('/plg_/', '', $plugin, 1));
```

Comment:
```suggestion
        $plugin = strtolower(preg_replace('/^plg_/', '', $plugin));
```
If it's really only the plugin name in variable `$plugin`, i.e. `'plg_something_plg_whatever'`, you could also use the begin of string anchor `^` in the regex. That would be more effective. However, if the value of that variable can have some kind of prefix before the name, it needs your solution.

Reply:
There are many ways to achieve the same thing. Not sure I understand "that would be more effective" ?

Reply:
Because it initially finds only the match at the beginning of the string. Using begin or end of string anchors make parsing a regex more effective.

Reply:
Hmm, of course a solution with substring like suggested by Fedir would be most effective. That should be preferred whenever possible to any regex operations.